hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/UpgradeSubcommand.java

```diff
@@ -0,0 +1,194 @@
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.ScmConfigKeys;
+import org.apache.hadoop.hdds.scm.cli.container.upgrade.UpgradeChecker;
+import org.apache.hadoop.hdds.scm.cli.container.upgrade.UpgradeManager;
+import org.apache.hadoop.hdds.scm.cli.container.upgrade.UpgradeUtils;
+import org.apache.hadoop.hdds.upgrade.HDDSLayoutFeature;
+import org.apache.hadoop.ozone.container.common.volume.HddsVolume;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+import java.io.File;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.Callable;
+
+/**
+ * This is the handler that process container upgrade command.
+ */
+@Command(
+    name = "upgrade",
+    description = "Upgrade schema v2 containers to v3",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class UpgradeSubcommand implements Callable<Void> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(UpgradeSubcommand.class);
+
+  @CommandLine.Spec
+  private static CommandLine.Model.CommandSpec spec;
+
+  @CommandLine.Option(names = {"--volume"},
+      required = false,
+      description = "volume name")
+  private String volume;
+
+  @CommandLine.Option(names = {"-y", "--yes"},
+      description = "Continue without interactive user confirmation")
+  private boolean yes;
+
+  private static OzoneConfiguration ozoneConfiguration;
+
+
+  @Override
+  public Void call() throws Exception {
+    final UpgradeChecker upgradeChecker = new UpgradeChecker();
+
+    OzoneConfiguration configuration = getConfiguration();
+
+    Pair<Boolean, String> pair =
+        upgradeChecker.checkDatanodeNotStarted();
+    final boolean dnNotStarted = pair.getKey();
+    if (!dnNotStarted) {
+      out().println(pair.getValue());
+      return null;
+    }
+
+    final Pair<Integer, Integer> layoutVersion =
+        upgradeChecker.getLayoutVersion(configuration);
+    final Integer softwareLayoutVersion = layoutVersion.getLeft();
+    final Integer metadataLayoutVersion = layoutVersion.getRight();
+    final int needLayoutVersion =
+        HDDSLayoutFeature.DATANODE_SCHEMA_V3.layoutVersion();
+
+    if (metadataLayoutVersion < needLayoutVersion ||
+        softwareLayoutVersion < needLayoutVersion) {
+      out().println(String.format(
+          "Please upgrade your software version, no less than %s," +
+              " current metadata layout version is %d," +
+              " software layout version is %d",
+          HDDSLayoutFeature.DATANODE_SCHEMA_V3.name(),
```

Comment:
Here print the HDDSLayoutFeature.DATANODE_SCHEMA_V3.layoutVersion() is preferred, to easy compare it with metadataLayoutVersion and softwareLayoutVersion. Or If we keep print the name, then, it's better to convert metadataLayoutVersion and softwareLayoutVersion name too.